packages/frontend-core/src/fetch/DataFetch.ts

```diff
@@ -44,22 +44,31 @@ interface DataFetchDerivedStore<TDefinition, TQuery>
   supportsPagination: boolean
 }
 
-export interface DataFetchParams<
-  TDatasource,
-  TQuery = SearchFilters | undefined
-> {
+export interface DataFetchParams<TDatasource, TQuery = SearchFilters> {
   API: APIClient
   datasource: TDatasource
   query: TQuery
-  options?: {}
+  options?: Partial<DataFetchOptions<TQuery>>
+}
+
+export interface DataFetchOptions<TQuery = SearchFilters> {
```

Comment:
since this is exported, would it be beneficial to put this in the types package?